EduNLP/Vector/gensim_vec.py

```diff
@@ -26,8 +28,28 @@ def __getitem__(self, item):
 
 
 class D2V(object):
-    def __init__(self, filepath):
-        self.d2v = Doc2Vec.load(filepath)
+    def __init__(self, filepath, method="d2v"):
+        self._method = method
+        self._filepath = filepath
+        if self._method == "d2v":
+            self.d2v = Doc2Vec.load(filepath)
+        elif self._method == "bow":
+            self.d2v = corpora.Dictionary.load(filepath)
+        elif self._method == "tfidf":
+            self.d2v = TfidfModel.load(filepath)
+        else:
+            pass
 
     def __call__(self, item):
-        return self.d2v.infer_vector(item)
+        if self._method == "d2v":
+            return self.d2v.infer_vector(item)
+        elif self._method == "bow":
+            return self.d2v.doc2bow(item)
+        elif self._method == "tfidf":
+            # 'tfidf' model shold be used based on 'bow' model
+            dictionary_path = re.sub(r"(.*)tfidf", r"\1bow", self._filepath)
```

Comment:
Modify this code segment. In current version, each `__call__` results in the initialization of bow model, which is time-consuming. Please write a new tfidf initialization (maybe you need to write a new tfidf class) to avoid this problem. In addition, if possible, please use a consistence inference function name like `infer_vector`. To do this, you may need to implement a new BoW class and Tfidf class, which in my opinion is a good solution.